### DISCUSSION TITLE
About the ASP conda environment and ISIS conda environment

### DISCUSSION THREAD

**Discussion by whigg** [NeoGeographyToolkit/StereoPipeline#377] [Q&A]
Hi,

I create an environment for asp and isis with conda seperately, but I am confused about the environment variable ISISROOT.

According to the ISIS installation instructions, set the ISISROOT environment variable:
export ISISROOT=$HOME/anaconda3/envs/isis 

According to the ASP installation instructions, set the ISISROOT environment variable:

export ISISROOT=$HOME/anaconda3/envs/asp

So, How exactly set the ISISROOT environment variables for these two virtual environments?

Thanks.

**Comment by rbeyer** [comment]
When you use conda to install ASP, the version of ISIS that ASP needs is installed in the *asp* conda environment that you have made.  You did not need to install ISIS separately.  So you now have two versions of ISIS on your system (which is no big deal).  

The instructions you point out for setting $ISISROOT for ASP are correct (if that's where your *asp* conda environment is installed: you used an anaconda3 install vs a miniconda install, and you named the environment "asp").

Please note that if you will be using ASP for planetary work (which requires ISIS), then you need to install an ISIS data area, so please follow the [ISIS instructions for doing so](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/README.md#the-isis-data-area).  Once it has been installed, then you'll also need to set the $ISISDATA environment variable, in addition to the $ISISROOT variable.  Section 7 of the [ISIS Installation with conda directions](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/README.md#isis-installation-with-conda) details the ways to do that.

**Reply by whigg** [reply]
Dear Ross Beyer,

Yes, I will be using ASP for planetary work. so I create ISIS and asp environment with conda.  I'm not sure what I understand is right about the $ISISROOT environment variable from  Section 7 of the [ISIS Installation with conda directions](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/README.md#isis-installation-with-conda). the environment variables ISIS requires should be set by
conda env config vars set ISISROOT=$CONDA_PREFIX ISISDATA=[path to data directory]
rather than export $ISISROOT to .bashrc. If $ISISROOT add to .bashrc,it will be conflict with the ASP environmental variable ISISROOT.

So, 
1. install isis as
conda create -n isis
conda activate isis
conda install -c usgs-astrogeology isis=7.0.0
setup the environment variables:
conda env config vars set ISISROOT=$CONDA_PREFIX ISISDATA=[path to data directory]

2.install asp as
conda create -n asp python=3.6
conda activate asp
conda install stereo-pipeline==3.1.0
then set the the environmental variable ISISROOT to .bashrc as
export ISISROOT=$HOME/anaconda3/envs/asp
export ISISDATA=[path to data directory]

Is this right?

Thanks very much!

**Comment by oleg-alexandrov** [comment]
The source of confusion is that ASP installed with conda already ships ISIS. You don't need to install ISIS separately. Use the one from $HOME/anaconda3/envs/asp. That one is ISIS 6.0.0. It has all the ISIS tools you may need. 

Now, if you do insist on using ISIS 7.0.0, which you install separately, then in any shell where you plan to run ASP tools, like parallel_stereo, set

export ISISROOT=$HOME/anaconda3/envs/asp

In any shell where you really want to run ISIS 7.0.0 tools, like cam2map, set:

export ISISROOT=$HOME/anaconda3/envs/isis

I will suggest you don't install ISIS 7.0.0 separately, unless you surely know that the version 6.0.0 shipped with ASP is not good enough for your needs.

**Reply by whigg** [reply]
Thanks oleg for your guidance, I guess I got your means. If I want to use ISIS 7 tools, in any shell, set:

export ISISROOT=$HOME/anaconda3/envs/isis

Now it is clearly that ASP installed with conda already ships ISIS, I will set the ASP environment according to your suggestions.

Thank very much for your guidance!
